src/v/syschecks/syschecks.cc

```diff
@@ -20,24 +20,56 @@
 #include <seastar/core/seastar.hh>
 #include <seastar/net/api.hh>
 
+namespace {
+ss::sstring to_string(ss::fs_type fs) {
+    return [fs]() {
+        switch (fs) {
+        case ss::fs_type::other:
+            return "other";
+        case ss::fs_type::xfs:
+            return "xfs";
+        case ss::fs_type::ext2:
+            return "ext2";
+        case ss::fs_type::ext3:
+            return "ext3";
+        case ss::fs_type::ext4:
+            return "ext4";
+        case ss::fs_type::btrfs:
+            return "btrfs";
+        case ss::fs_type::hfs:
+            return "hfs";
+        case ss::fs_type::tmpfs:
+            return "tmpfs";
+        };
+        return "bad_enum";
+    }();
+}
+} // namespace
+
 namespace syschecks {
 ss::logger checklog{"syschecks"};
 
 ss::future<> disk(const ss::sstring& path) {
     return ss::check_direct_io_support(path).then([path] {
         return ss::file_system_at(path).then([path](auto fs) {
-            if (fs == ss::fs_type::ext4) {
+            checklog.info0("Detected file system type is {}", to_string(fs));
+            // all of ext2, 3 and 4 are detected as ext2, so we just assume an
+            // ext2 detection means ext4 for now, see:
+            // https://github.com/redpanda-data/redpanda/issues/13469
+            if (fs == ss::fs_type::ext2) {
```

Comment:
Shouldn't we just also make this check for ext3 and ext4? Then it will automatically work once upstream is fixed.

Reply:
+1 perhaps we should explicitly define the set of supported filesystems and the set of recommended filesystems, and log them each in these warning/error messages

Reply:
@StephanDollberg - I'm not sure. If we get `ext3` as a response, what does it mean? That upstream has been fixed and the fs being is used is actually ext3? Officially we don't support ext3, and so I'm not sure the warning is right in spirit and it's also wrong when it says "you are on ext4 ...". Similarly if upstream has been fixed, then it's wrong to include ext2 in this check: it should be changed to only include ext4 at that point.

Reply:
However, I agree that if we ever get `ext4` I guess we should print this ext4 warning. So maybe `fs == ext2 || fs == ext4` is the best check? Seems weird.

Reply:
BTW here's my suggested upstream fix:

https://github.com/scylladb/seastar/pull/1837

It does not change the behavior of the existing method, but adds a new one. So in that case we'd need another change here to use the new method. I'm not sure if it will be accept or if it's the best approach.

Reply:
> +1 perhaps we should explicitly define the set of supported filesystems and the set of recommended filesystems, and log them each in these warning/error messages

As far as I know there is only one supported and recommended file system: xfs. ext4 is in a grey zone.

In any case, does the code not explicitly define these now via its "if cascade"?

Reply:
Right, from what you mentioned in the meeting I had understood that ext2/3/4 are actually quite similar.

> However, I agree that if we ever get ext4 I guess we should print this ext4 warning. So maybe fs == ext2 || fs == ext4 is the best check? Seems weird.

This does seem like a good middle ground even if a bit weird.

I guess it all depends a bit on what you want to do with the seastar fix. If you are going to cherry pick it into our fork now anyway then yeah it's fine lets just merge as is. Just want to avoid a scenario where we later update seastar which fixed the detection and then this breaks. Backports also an angle to consider but I guess we never really pull in new stuff there (unless done intentionally).

Reply:
> I guess it all depends a bit on what you want to do with the seastar fix. If you are going to cherry pick it into our fork now anyway then yeah it's fine lets just merge as is.

I don't plan to cherry-pick it immediately but see what happens upstream. I don't have any strong expectation there: it's entirely possible nothing gets accepted and so the current state of the syscheck lives forever, or maybe it will be changed to not introduce a new method but actually fix `file_system_at()` in which case we would suddenly start returning `ext3` or `ext4` here as you are concerned about.

In any case, I agree that the `== ext2 | ext4` check is reasonable and I'll change it to that.

> Backports also an angle to consider but I guess we never really pull in new stuff there (unless done intentionally).

Do you mean backporting this change? I plan to backport it to 23.2 only.

Reply:
> Do you mean backporting this change? I plan to backport it to 23.2 only.

Yeah meant this change. Sounds fine, probably no new clusters being set up on 23.1 so unlikely to run into the confusing log message.